### PR TITLE
Added Madeline

### PR DESCRIPTION
**Pull request by NCKiser**
Added Madeline keyboard, default and vial keymaps
PCB by Rossman360, firmware ported to QMK 1.1.1 / Vial 0.7 by NCKiser/kiserdesigns
Case and PCB Group-Buy by rubybuilds

Tested both keymaps on actual hardware.
More kiserdesigns keyboard to come in the near future.

### CODE REVIEW

File: keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+  [7] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (lesshonor) [inline keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c]: No need to hardcode filler `KC_TRNS` layers. Dynamic keymaps will automatically fill in "missing" layers at compile-time up to the maximum set in `dynamic_keymap.layer_count`.
```suggestion
```

File: keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =   { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN) },
+    [1] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [2] =   { ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [3] =   { ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [4] =   { ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [5] =   { ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [6] =   { ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [7] =   { ENCODER_CCW_CW(KC_NO, KC_NO) }
```
Comment (lesshonor) [inline keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c]:
```suggestion
    [1] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
    [2] =   { ENCODER_CCW_CW(KC_NO, KC_NO) }
```

File: keyboards/kiserdesigns/madeline/keymaps/vial/rules.mk [hunk 3]
```diff
@@ -0,0 +1,7 @@
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+COMBO_ENABLE = yes
+TAP_DANCE_ENABLE = yes
+VIAL_ENCODERS_ENABLE = no
+ENCODER_MAP_ENABLE = yes
+VIAL_INSECURE = no
```
Comment (lesshonor) [inline keyboards/kiserdesigns/madeline/keymaps/vial/rules.mk]:
```suggestion
ENCODER_MAP_ENABLE = yes
```
The rest are on by default or no longer needed.

File: keyboards/kiserdesigns/madeline/keymaps/vial/vial.json [hunk 4]
```diff
@@ -0,0 +1,189 @@
+{
+    "name": "Madeline",
+    "vendorId": "5242",
+    "productId": "6D64",
```
Comment (lesshonor) [inline keyboards/kiserdesigns/madeline/keymaps/vial/vial.json]:
```suggestion
```
You can leave them if you want, but these technically aren't necessary.

**Review by lesshonor — CHANGES_REQUESTED** [review]

File: keyboards/kiserdesigns/madeline/keymaps/vial/vial.json (hunk 4 above)
Comment (NCKiser) [reply keyboards/kiserdesigns/madeline/keymaps/vial/vial.json]: I honestly thought they were needed. I always forget to update the vid and pid in vial.json when writing new firmware, glad to know they can be omitted!

File: keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c (hunk 2 above)
Comment (NCKiser) [reply keyboards/kiserdesigns/madeline/keymaps/vial/keymap.c]: Turns out the encoder and keymap layers defined still need to match, even if they are both padded by Vial

**Review by lesshonor — APPROVED** [review]
Builds for me.

**Pull request merged**